### DISCUSSION TITLE
Write documentation to integrate with new build backends besides setuptools

### DISCUSSION THREAD

**Discussion by Ambro17** [pallets/click#2191] [Ideas]
Today setuptools is being replaced by new PEP standards and single purpose libraries that implement them. `build` is one case to build sdists and wheels, flit is another case.

Nowadays one doesn't require setuptools in order to setup entrypoints, flit allows that too. Would you accepto a documentation PR that adds instructions? Or is click's use case too complex that it depends on any feature only by setuptools that would make suggesting other tools harmful? 

Thanks in advance!

**Comment by davidism** [comment]
I'll accept something that shows [`pyproject.toml` `[project.scripts]`](https://www.python.org/dev/peps/pep-0621/#entry-points) use. I don't want to reference any tool-specific configs besides setuptools though. All tools should be standardizing on that. Entry points are a standard packaging spec, not setuptools specific.

**Reply by Ambro17** [reply]
Great to know, do you want me to open a PR once i have it ready or first an issue to keep things organised?

**Reply by risoms** [reply]
@Ambro17 @davidism Just wanted to follow up - Has there been any work on this?
